portfolio/src/main/java/com/google/sps/servlets/DeleteServlet.java

```diff
@@ -26,11 +31,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     System.out.println("Button val: " + id);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Query query = new Query("Task").setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
-    Entity entity = results.asSingleEntity();
-    datastore.delete(entity.getKey());
-    
+
+    for (Entity entity : results.asIterable()) {
+      long comment_id = (long) entity.getProperty("comment_id");
+
+      if (Long.toString(comment_id).equals(id)) {
```

Comment:
A NullPointerException would be thrown if the comment_id is null. So it is better to prevent it in advance.